### PR TITLE
Kaitie edit section UI modifications

### PR DESCRIPTION
**Pull request by kobryan0619**
This does a lot of things.   It:

- creates a prop that will change the UI of the landing page depending on if it is coming from a new section or an existing section.
- pops open the right view for existing courses based on audience
- Makes the Advanced Settings Responsive to the course choice
- Moves the existing data about the course into the "edit section" UI.
- Blocks out Advanced Settings into some dividers 

If the prop gets passed is as NOT a new section, people will see this UI:
image

## Links

- [Spec](https://docs.google.com/document/d/1qZrH_k7iLqqWA54w0aDtABudNa0QrGRMe4gek1se7r8/edit#)

## Testing story

- I added one test to check that the header and a button were still showing up.
- I visually checked the data for sections with HOC, Elementary, Stand-alone, full courses w/o a unit assigned, courses with a specific unit assigned.  I also checked that this worked for different versions other than the current version assigned and that this didn't break anything for new section sign-up flow. 
- We still need to add tests.  Right now, I am looking for review on the new code.  

## Follow-up work

- Add tests
- Refactor the "data blob" to be more efficient
- Add in PL

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: tiny nit: do you mind moving this up to line 2? We're not good at ordering imports here but generally, the React-related imports are first in the list. Don't block this PR on this though!

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: nit: with functional components you can destruct props in the parameter passed to the function, ie
```
export default function SectionsSetUpContainer({isNewSection}) {
```

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: I'm curious about what the plan is for piping the section to be edited through this component? In particular, is it worth instead having a prop like `sectionToBeEdited` and, if that's null, assume that we're creating a new section?

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx]: I like how that reads better.  I added this to the file and changed the tests for it.

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:21]: Ayyy I'm happy to see this switched over!

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:177]: I like the move to only display this button for new sections🎉

File: apps/test/unit/templates/sectionsRefresh/SectionsSetUpContainerTest.jsx
Comment (hannahbergam) [inline apps/test/unit/templates/sectionsRefresh/SectionsSetUpContainerTest.jsx]: I'm sure you're working on this part still, but just a visual reminder to edit now that the input is no longer a boolean!

File: apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx:34]: was this breaking something? the api should support the new `grades` param for this

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx:34]: It was, for the the "edit section" screen, it was looking at the data of an existing section.  The existing section had the param named "grade" not "grades".   I noticed this when testing the `SectionsSetupContainer` for the edit page. 

I do not have experience with the api you mentioned... should that also work for existing sections?

Comment (rshipp) [reply apps/src/templates/sectionsRefresh/SingleSectionSetUp.jsx:34]: ohh, that makes sense. i wonder if that's something I missed when I migrated things over to support `grades` in #50309. if having it as `grade` makes this work for new and existing sections though, that's fine.

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (kobryan0619) [inline apps/src/templates/sectionsRefresh/QuickAssignTable.jsx:1]: @rshipp can you take a look at this solution for updating the `selectedCourseOffering`?  I am in particular interested in:

1. Efficiency
2. Use of the "dependencies" parameter - I haven't used this so far and reading the documentation I cannot tell if this is an indicator that something could be written better OR, if this is the best way to go about it
3. Location of this useEffect.  I tried moving it into the parent component where the states are being create used, but got all sorts of errors.

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: nit: since this is calling a function, and not returning anything, i think it would be a little clearer to use a regular loop and `if`, like:
```suggestion
        courseDataByHeaderValues.forEach(course => {
          if (sectionCourse?.courseOfferingId === course.id) {
            setSelectedCourseOffering(course);
          }
        });
```
same for the `headers.map` above, too. this might just be my personal preference, but when I see `map`, i tend to assume we're trying to modify and return data formatted in a different way, where `forEach` makes it immediately clear we don't care about a return value and are just doing things inside a loop.

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: this seems like a great way to avoid having to recurse over the whole `courseOfferings` object!

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: the dependencies list should include any variables that, when they are changed, you want the component to re-render and to run the useEffect function again. since `setSelectedCourseOffering` is a function, it shouldn't need to be in the dependencies. without reading through this component in detail, I'm not sure about the others.

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (rshipp) [inline apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: this seems like an ok place to put the `useEffect` to me? i am curious what broke when you put it in the parent component, but I don't see a strong reason that makes me think it should be in one place over the other

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: Thanks, I really appreciate the explanation behind this proposed change.

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: My brain is still wishing we could find a way to initialize the selectedCourseOffering with this data a level higher, but I agree that having a marketing audience at this point helps narrow down the offerings. But now I'm curious- does this still set the correct selected element if the user opens a different marketing audience first than the one they've already got an assigned course in? For example: if a user has CSD assigned but first opens the elementary school dropdown, what happens?

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: I think this matches my mental model too^ :)

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (rshipp) [reply apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: >  since `setSelectedCourseOffering` is a function, it shouldn't need to be in the dependencies

just saw some code that used a function in the dependencies, so I might be wrong about this! i'd be interested to see what you end up with, since i know you were looking into useEffect more

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx]: You can probably do something like `courseVersion.content_root_type` here (but need to check on the exact property name)

Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx]: Oo good point- and even clearer now that we've deleted all content_root_type = 'script'😂. Kaitie- the content root types are 'Unit' and 'UnitGroup'. UnitGroups are for courses, and Unit indicates a standalone. Just a simpler way to do the same thing!

File: apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx:97]: Love these name clarifications/simplifications!

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx]: Just calling out a couple comments to clean up!

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx]: I think we could refactor to use a flatten-like function (maybe in a helper file? We might have use for something like that elsewhere). Can be followup work

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx]: We are now in the useEffect:)

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:70]: I think this helper function could be pulled outside of the useEffect so that the function a little shorter/easier to read. Up to you

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx]: I think you could consolidate the two updateCourse calls by using 'let' for hasLessonExtras and hasTextToSpeech, assigning them if standalone, and setting them in the updateCourse either way (to null if we don't know) for a shorter/cleaner function. Then just double check that the advanced settings usage will ignore them if they exist but are null.

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx]: Curious about why we needed to pass 'isNewSection' through!

**Review by hannahbergam — COMMENTED** [review]
Getting so close! Great work!

File: apps/src/templates/sectionsRefresh/QuickAssignTable.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/QuickAssignTable.jsx]: Thanks!  Yes, I was getting a linting error when I tried committing without the function in the dependencies.  Bethany said in passing that she didn't love our linter for this reason, but I'd love to learn more about her thoughts on that so I can understand the dependencies more.  It is weird because it seems to work without dependancies listed.  I probably need to do a bit more self-study on this too, to understand the why behind it all.

**Review by kobryan0619 — COMMENTED** [review]
Made changes documented in the description

File: apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/VersionUnitDropdowns.jsx]: Yep - this doesn't need to be here!

File: apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx]: is lessonExtras ever undefined? I am wondering if an || would work here

Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/AdvancedSettingToggles.jsx]: Additionally, you can use the '?' instead of checking if something is equal to undefined!

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:45]: Thanks for adding this comment! This could be included explicitly in a followup ticket that you or I (or even someone else on the team) could pick up soon

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx]: Does this still need to be here?

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:102]: Nice refactor here🎉

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:145]: Oh nice! Looks like you're tackling https://codedotorg.atlassian.net/browse/TEACH-421 right here. Any chance you can link this ticket: https://codedotorg.atlassian.net/browse/TEACH-278 and this pr so we can find what the button should look like when we're reenabling?

**Review by hannahbergam — COMMENTED** [review]
Huge progress! Mostly requests for tracking followup work. I know you did a ton of thinking/manual testing around different kinds of sections for this- can you track those learnings/scenarios in a followup testing ticket?

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:45]: Created in [TEACH-438](https://codedotorg.atlassian.net/jira/software/projects/TEACH/boards/65?selectedIssue=TEACH-438)

Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:45]: Added!

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx]: Nope - deleting!

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:145]: Will do!  I already got TEACH-421.  I did comment this out in the newest commit.  I will note it int TEACH-278 too.

Comment (hannahbergam) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:145]: Yesss thanks!

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:70]: I did end up pulling out a different function from this useEffect to make it easier to read (`updateSectionCourseForExisitngSections`).

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:76]: Leaving a note here for whoever ends up doing this refactor work: this and the call below could at the very least be cut off once a selectedCourseOffering is assigned (preventing a search through all the offerings every time).

**Review by hannahbergam — COMMENTED** [review]
Okay I think we've caught all the future work and made update tickets but I might be too close to it at this point 😂. Just so I know- this doesn't address PL yet. Once this is merged, what happens if you open the page with audience = teachers?

**Comment by cat5inthecradle** [comment]
This PR is huuuuuuuge... Is there any way to break it up in to smaller changes?

I'm also seeing some likely unintentional edits to files like the pull_request_template.md or CONTRIBUTING.md. Looks like that may be related to merge conflicts.

**Comment by cat5inthecradle** [comment]
Look, I saw you were deleting hundreds of thousands of files and I got too hopeful :laughing:

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:68]: possibly as a follow up, do you also need to expand the PL offerings?

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx
Comment (bethanyaconnor) [inline apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:91]: Are these options not supported for non-standalone units?

File: dashboard/app/controllers/sections_controller.rb
Comment (bethanyaconnor) [inline dashboard/app/controllers/sections_controller.rb:14]: I don't think this is needed for edit section. `loginType` and `participantType` should already be set for existing sections

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:68]: Yes - this is on my "next" list.  Right now we wanted to get this out the door so Hannah could work on her part a bit more easily.

File: apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx:91]: They are.  The issue is that once you choose "Stand alone Unit", the computer should know at that point if these two items are available.  For all other material, you need to have a chosen unit to know if they are available.  This is why this is set at this point for standalone units

File: apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx
Comment (kobryan0619) [reply apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx:68]: That's the plan!

File: dashboard/app/controllers/sections_controller.rb
Comment (kobryan0619) [reply dashboard/app/controllers/sections_controller.rb:14]: Agreed.  Getting rid of this will cause some issues for another file expecting the param (specifically, the `useEffect` in `CurriculumQuickAssign` I am going to make a ticket for this for us to address separately.

Comment (kobryan0619) [reply dashboard/app/controllers/sections_controller.rb:14]: Noting that this is now https://codedotorg.atlassian.net/browse/TEACH-447

**Review by hannahbergam — APPROVED** [review]
Happy to see this in a mergeable state! Great perseverance!

**Pull request merged**
